devito/types/array.py

```diff
@@ -203,6 +211,13 @@ def _mem_shared(self):
     def _mem_constant(self):
         return self._scope == 'constant'
 
+    @property
```

Comment:
Move to `AbstractFunction`  I'd say

Reply:
I will see how I can improve things. Note that DiscreteFunction has a `_distributor` attribute too

Reply:
I think so yes

Reply:
done